parsers/PH.py

```diff
@@ -0,0 +1,696 @@
+        "datefilter%5Bend%5D": "2021-09-13+12%3A20",
+        "sort": "",
+        "page": "1",
+        "post_id": KIND_TO_POST_ID[kind],
+    }
+    r = requests.post(
+        "https://www.iemop.ph/wp-admin/admin-ajax.php", data=form_data, verify=False
+    )
+    id_to_items = eval(r.text)["data"]
+    datetime_to_items = {}
+    for id, items in id_to_items.items():
+        market_reports_item = MarketReportsItem(
+            datetime.strptime(items["date"], "%d %B %Y %H:%M"),
+            items["filename"],
+            DIPC_URL + f"?md_file={id}",
+        )
+        datetime_to_items[market_reports_item.datetime] = market_reports_item
+    logger.info(f"PH - {kind}: Succesfully recovered market reports items")
+    return datetime_to_items
+
+
+def filter_reports_items(
+    reports_items: List[MarketReportsItem], target_datetime: Optional[datetime]
+) -> List[MarketReportsItem]:
+    # Date filtering
+    if target_datetime is None:
+        last_available_datetime = max(reports_items.keys())
+        start_datetime = last_available_datetime - timedelta(days=1)
+        reports_items = [
+            item for item in reports_items.values() if item.datetime >= start_datetime
+        ]
+    else:
+        # Refetch the whole day
+        reports_items = [
+            item
+            for item in reports_items.values()
+            if item.datetime.date() == target_datetime.date()
+        ]
+    return reports_items
+
+
+def download_production_market_reports_items(
+    session: Session, reports_items: List[MarketReportsItem], logger: Logger
+) -> pd.DataFrame:
+    from io import BytesIO
+    from zipfile import ZipFile
+
+    COLUMNS_MAPPING = {
+        "REGION_NAME": "zone_key",
+        "RESOURCE_NAME": "resource_name",
+        "SCHED_MW": "production",
+    }
+
+    _all_items_df = []
+    for reports_item in reports_items:
+        logger.debug(f"Downloading market reports for {reports_item.filename}")
+        res: Response = session.get(reports_item.link)
+        _item_dfs = []
+        # zip containing a list of csv files which we want to concatenate in a single dataframe
+        with ZipFile(BytesIO(res.content)) as zip_file:
+            for csv_filename in zip_file.namelist():
+                if csv_filename.endswith(".csv"):
+                    with zip_file.open(csv_filename) as csv_file:
+                        _df = pd.read_csv(csv_file, header=0)
+                        # The last line is EOF
+                        _df = _df[:-1]
+                        # All numeric types
+                        _df = _df.apply(pd.to_numeric, errors="ignore")
+                        # Add datetime column
+                        _df = convert_column_to_datetime(_df, "TIME_INTERVAL")
+                        # Localize to TIMEZONE
+                        _df["datetime"] = _df["datetime"].apply(
+                            lambda x: x.tz_localize(TIMEZONE)
+                        )
```

Comment:
Perhaps we want to do the conversion to datetimes here with pd.to_datetime() instead? Would solve the below type error as well.